### ISSUE TITLE
Nextcloud Client overwrites all files on server after power outage

### ISSUE THREAD

**Issue by Rocreex** [nextcloud/server#5965]
The error is hard to reproduce, however I am going to explain what happened:
After a power outage and restarting the computer, the official Nextcloud client tried to reupload ALL data, which it synchronizes. I am not quite sure whether the outage caused the operating system to mess with the files or the "modified" timestamps.
The server in contrast shows a huge list of error messages:
"Sabre\DAV\Exception\MethodNotAllowed: HTTP/1.1 405 The resource you tried to create already exists"
So the server obviously was aware of the client trying to upload files which are already on the server. Nevertheless, the server shows new versions for every file, so even though the server generated an error, it still somehow updated the files? The files, by the way, have not changed.

The question is: How do server or client check whether the file has actually been modified? Is only the timestamp considered? Does the client do hashing in any way?

I would expect either the client or the server to recognize that there has been no change to the files. 

The client used is the lates stable Nextcloud client and the server is also the latest stable release (12.0.0).

**Comment by MorrisJobke** [comment]
There is a etag stored for every file/folder. That etag changes if the file has been modified (but it's not a hash of the file, but only needs to change to something else if the file has changed). Then the client knows, because it stores the etag of all files that was reported on the last sync run, which files to update. If the server restored an old version you need to set a special flag in the config.php to indicate this to the clients: 

https://github.com/nextcloud/server/blob/master/config/config.sample.php#L1518-L1531


@rullzer @pfiff45 Did you ever noticed something like that?

**Comment by rullzer** [comment]
I did not. My bet is something (maybe a DB) got corrupted. But since that is our source of truth I see no solution.

**Comment by Isvvc** [comment]
I appear to be having a similar issue, though not as a result of a power outage or similar. I have about 24 GiB of new data I've been attempting to sync, but the transfer speeds with the desktop client have been extremely slow, so I manually transferred the files from the server to my PC over SFTP into the corresponding location in my local Nextcloud Sync folder, but the Nextcloud client seems to be trying to rewrite all of those files, despite them seemingly being exactly the same as what's on the server.

**Comment by skjnldsv** [comment]
As I cannot reproduce the original issue anymore, I will close this ticket. If this is still happening please make sure to **upgrade** to the **latest** version. After that, feel free to reopen.

**Comment by thebeline** [comment]
I hate to revive a dead topic, but I seem to have had a similar issue:

We had a power outage, and following that, the server started throwing `503` errors, saying that file `X` could not be found.

Looking on the We interface for a specific example, I could see that the folder contained one file of a name, let's say `A`.  Which was at one point (over a month ago), the only file in that folder.

However, looking at the system folder via SSH shows two files `B` and `C` which were created only a week and a half ago.  The web interface is trying to sync the old file `A`, and does not seem to recognize that `A` was removed, and `B` and `C` should be synced.

At this same time, the NC windows client removed `B` and `C` from my local folder, and the folder is now empty (as `A` could not sync in).

Very curious.

Additionally, some other files, such as a few very large Lightroom catalogs, and numerous KeePass files (the only examples I have found so far) no longer open and state they are corrupted.

I am currently running `occ files:scan` to discover the "missing" files, and I HOPE that when the re-sync occurs, it will fix the "corrupted" files, but I don't have a lot of faith...  Something very bad happen, and I don't know what...

Incidentally, the modified times for the corrupted files look correct (not changed) on local, web, and the server.  So...  :-/
